stan/math/prim/mat/fun/matrix_exp_2x2.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_MATRIX_EXP_2X2_HPP
+#define STAN_MATH_PRIM_MAT_FUN_MATRIX_EXP_2X2_HPP
+
+#include <unsupported/Eigen/MatrixFunctions>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+
+namespace stan {
+    namespace math {
+
+        /**
+         * Return the matrix exponential of a 2x2 matrix. Reference for 
+         * algorithm: http://mathworld.wolfram.com/MatrixExponential.html
+         *
+         * @param[in] A A 2x2 matrix
+         * @return Matrix exponential of A.
+         */
+        template <typename T>
+        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+        matrix_exp_2x2
+          (const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& A) {
```

Comment:
Put this all on one line---lines should go out to 80 characters max.  If that won't fit, put the open paren `(` right after function and put argument on next line.  But I'm pretty sure this one will fit.